### PR TITLE
H-2915, H-2917: Show specific task in cost breakdown and LLM logs

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

Adds more detail to help with development:
1. Store specific task (e.g. facts from text, deduplicate entities) and Flow step id on usage record
1. Show cost breakdown per task in UI
1. Add task name and step id to LLM debug logs

Driveby:
- Fix an issue with the AI assistant not being able to create the `incurredIn` link between the usage record and the Flow (because it no longer has view permissions on the usage record itself)

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

## ❓ How to test this?

1. Wipe db and regenerate types
2. Run a Flow, check UI and logs

### CODE REVIEW

File: apps/hash-frontend/src/pages/[shortname]/shared/flow-visualizer/topbar.tsx [hunk 1]
```diff
@@ -191,7 +191,10 @@ export const Topbar = ({
                 >
                   Run
                   {run.closedAt
-                    ? ` – ${format(new Date(run.closedAt), "yyyy-MM-dd h:mm a")}`
+                    ? ` – ${format(
+                        new Date(run.closedAt),
+                        "yyyy-MM-dd h:mm a",
+                      )}`
```
Comment (benwerner01) [inline apps/hash-frontend/src/pages/[shortname]/shared/flow-visualizer/topbar.tsx:197]: I'm curious where these diffs are coming from, I don't see any changes to prettier in this PR - is it coming from your editor?

**Review by benwerner01 — APPROVED** [review]
Thanks Ciaran, looks good and is working locally for me as expected 👍

File: apps/hash-frontend/src/pages/[shortname]/shared/flow-visualizer/topbar.tsx (hunk 1 above)
Comment (CiaranMn) [reply apps/hash-frontend/src/pages/[shortname]/shared/flow-visualizer/topbar.tsx:197]: I don't know, I tried to figure this out - linting is passing and formatting should be deterministic... I also didn't touch or open some of these files. There was a recent patch prettier update, dunno if that's done something.

I also get warnings on prettier lint locally now and AFAIR we aren't supposed to have warnings

Comment (CiaranMn) [reply apps/hash-frontend/src/pages/[shortname]/shared/flow-visualizer/topbar.tsx:197]: Will see if it starts happening in other PRs and look further if so

**Pull request merged**